CMakeLists.txt

```diff
@@ -460,8 +453,8 @@ if (gflags_FOUND)
   endif (NOT BUILD_SHARED_LIBS)
 endif (gflags_FOUND)
 
-set_target_properties (glog PROPERTIES VERSION ${GLOG_MAJOR_VERSION})
-set_target_properties (glog PROPERTIES SOVERSION ${GLOG_VERSION})
+set_target_properties (glog PROPERTIES VERSION ${PROJECT_VERSION})
+set_target_properties (glog PROPERTIES SOVERSION ${PROJECT_VERSION_MAJOR})
```

Comment:
I'm not sure this is correct. Does this mean there will be only a `libglog.so.0` but not a `libglog.so.0.3.5`?

Reply:
No there will be 3 files: (-> means symbolic link aka ln -s ...)
libglog.so -> libglog.so.0 (since SO_VERSION which is the API compatibility is only major)
libglog.so.0 -> libglog.so.0.3.5 (so the "API" point on your build version)
libglog.so.0.3.5 (which is the "real" library)

As I said in the commit message, there is also an inversion between VERSION and SOVERSION usage
(easy to spot with make install DESTDIR=install && ls -l install/use/local/lib)
doc:  https://cmake.org/cmake/help/latest/prop_tgt/VERSION.html#prop_tgt:VERSION

Hope it's clear know :sweat_smile: 

side note: if you wrap your code in a python "setup.py" bdist will resolve the symlink so you just have to depend on `$<TARGET_SONAME_FILE:glog>`
cf https://github.com/google/or-tools/blob/mizux/dev/cmake/python.cmake#L110